src/inference/graph_inference.cpp

```diff
@@ -1101,10 +1190,46 @@ void GraphInference::SSVMLearn(
       VLOG(3) << a->GetLabelName(it->first.a_) << " " << a->GetLabelName(it->first.b_) << " " << a->GetLabelName(it->first.type_) << " " << it->second;
       auto features_it = features_.find(it->first);
       if (features_it != features_.end()) {
-        features_it->second.atomicAddRegularized(it->second, 0, svm_regularizer_);
+        features_it->second.atomicAddRegularized(it->second, 0, regularizer_);
+      }
+    }
+  }
+}
+
+void GraphInference::PLLearn(
+    const Nice2Query* query,
+    const Nice2Assignment* assignment,
+    double learning_rate,
+    PrecisionStats* stats,
```

Comment:
In this comment I am a bit confused what are you referring to when you say unused?